spring/src/test/java/io/kubernetes/client/spring/extended/controller/KubernetesReconcilerCreatorTest.java

```diff
@@ -96,10 +127,53 @@ private boolean podInformerCacheReady() {
     }
   }
 
-  @Autowired private Controller testController;
+  @Resource(name = "test-reconcile")
+  private Controller testController;
+
+  @Resource private TestReconciler testReconciler;
+
+  @Resource private SharedInformerFactory sharedInformerFactory;
+
+  public static class CustomWorkQueueKeyFunc implements Function<KubernetesObject, Request> {
+
+    public CustomWorkQueueKeyFunc(WorkQueue<Request> workQueue) {
+      this.workQueue = workQueue;
+    }
+
+    private final WorkQueue<Request> workQueue;
+
+    @Override
+    public Request apply(KubernetesObject item) {
+      workQueue.add(new Request("foo"));
```

Comment:
@tony-clarke-amdocs here is an example of customized logic for pushing requests to the queue

Reply:
So we just need to implement WorkQueueKeyFunc that maps back to the name of primary resource?

Reply:
there's two separate cases you mentioned in the issue:

- __mapping 1 secondary resource to 1 primary resource__: e.g. referencing by owner-reference. i think this is already supported and you can simply make that work by implementing `WorkQueueKeyFunc` which for example craft a controller-request by unpacking a pod's owner-reference.

- __mapping 1 secondary resource to multiple primary resources__: this case is what this pull trying to address. for now the `WorkQueueKeyFunc` only supports mapping one resource to one request b/c it is declared to be `Function<KubernetesObject, Request>`. in order to achieve that, this pull allows the `WorkQueueKeyFunc` to return a `null` request to indicate the framework not to push anything to the queue, and instead the `WorkQueueKeyFunc` craft/pushes the requests to the queue directly. does that make sense to you?

Reply:
Thanks! I didn't realize I could already do use case one with the WorkQueueKeyFunc...I will try that out. I understand what I need to do to support the second scenario...maybe we need to add some javadoc to explain how a user is expected to return null and add their own entries to the queue. What version might this be available in?